### ISSUE TITLE
Add license

### ISSUE THREAD

**Issue by kinow** [cylc/cylc-uiserver#22]
No license added yet.

@cylc/core GPL v3 like Cylc I assume?

**Comment by kinow** [comment]
And up for grabs if anyone would like to submit the LICENSE.txt. Bonus points if you

- updates `setup.py` ([isodatetime](https://github.com/metomi/isodatetime/blob/b6a0eaf091c68ba2f4826fefbe69bbc35f52a747/setup.py#L76) has a good example)
- adds the license header to the project files

:grin:

**Comment by hjoliver** [comment]
Yes, same license as currently.

**Comment by dwsutherland** [comment]
I just imported [graphene-tornado](https://github.com/graphql-python/graphene-tornado) into our code (MIT), temporarily as I need a change in there next release (2.0.2).. I added a licence folder:
https://github.com/dwsutherland/cylc-uiserver/tree/ws-graphql-integration

Right thing to do?

**Comment by dwsutherland** [comment]
Should put contributors file in also.

**Comment by hjoliver** [comment]
I'm not sure bringing graphene-tornado into our code base is the right thing to do. Can you not just use a separate local modified copy of it?  (I realize that's more painful to use, but we've had enough of bundled 3rd party packages now ... Jinja2, cherrypy...)

**Comment by dwsutherland** [comment]
> I'm not sure bringing graphene-tornado into our code base is the right thing to do. Can you not just use a separate local modified copy of it? (I realize that's more painful to use, but we've had enough of bundled 3rd party packages now ... Jinja2, cherrypy...)

You mean; modify the previous release on install?
I haven't changed their code, just waiting for the next release.. Yeah Ok I'll just do that.

**Comment by kinow** [comment]
@dwsutherland we can use `setup.py` to get a copy of that branch without adding it to our code base. Do you have a branch with your code? I can give it a try and send a pull request.

**Comment by dwsutherland** [comment]
> @dwsutherland we can use `setup.py` to get a copy of that branch without adding it to our code base. Do you have a branch with your code? I can give it a try and send a pull request.

Oh, we don't need to modify their code, just clone/take from the HEAD of their project:
https://github.com/graphql-python/graphene-tornado

I'm redoing my branch now.

**Comment by kinow** [comment]
>Oh, we don't need to modify their code, just clone/take from the HEAD of their project:

But with `setup.py`, if you don't need to modify the code, then you don't even need to clone it. Right now we have something like this in `setup.py`:

```
...
setup(
  ...,
  install_requires = [
    'graphene-tornado'
 ]
)
...
```

I don't recall how's the syntax exactly (or if we need to write a short Python code) but we can do something like:

```
...
setup(
  ...,
  install_requires = [
    'https://github.com/graphql-python/graphene-tornado/commit/c24e23a1a62855e56c6ab0b6e55addf7e69904a0'
 ]
)
...
```

Then doing `pip install .` or `pip install -e.` will download all project dependencies, just like before, except that instead of `graphene-tornado`, it will clone the `graphene-tornado` from that commit, run `pip install blablabla`, and then you can safely import from `graphene_tornado` and use the code from that commit :tada:

**Comment by kinow** [comment]
But if it's temporary and you don't need that now (i.e. not planning to send a pull request with the code), then we can just wait for them to merge it.

**Comment by dwsutherland** [comment]
I'll just modify my virtualenv for now... When I do create a pull request, then we can reassess.

**Comment by hjoliver** [comment]
(I misread above, I thought or assumed? you said you needed to modify their code @dwsutherland )

**Comment by dwsutherland** [comment]
> (I misread above, I thought or assumed? you said you needed to modify their code @dwsutherland )

Only if we went with their `2.0.1` release...
BTW there's also a conflict in their Jinja2 requirements and what's installed with jupyterhub, oddly enough. It's probably used with jupyterhub's tornado server, and graphene-tornado could only use it for the graphiql interface template rendering (I think))..
graphene-tornado have is Jinja2==2.9.6 but Jupyter has no restrictions...  (maybe back compat reasons?)

Anyways, this causes a failure on start-up in the package checker:
> pkg_resources.DistributionNotFound: The 'Jinja2==2.9.6' distribution was not found and is required by graphene-tornado

but I changed it to `Jinja2>=2.9.6` in my virtualenv ... no issues with `2.10.2`

**Comment by kinow** [comment]
Had noticed a warning during build, but it seemed to work. Good catch @dwsutherland , let's work on the fix on #28.
